File: apple/Sources/CCDAEngine/Models/Media/CCDAMediaPayload.swift
import Foundation

/// Media payload storage.
public enum CCDAMediaPayload: Hashable, Sendable {
    /// Payload decoded and written to disk.
    case cachedFile(URL, byteCount: Int)
    /// Payload kept as base64 text, usually when caching is disabled or decoding fails.
    case inlineBase64(String)
    /// Payload was not available or could not be represented.
    case unavailable

    /// Cached file URL when payload is cache-backed.
    public var fileURL: URL? {
        if case .cachedFile(let url, _) = self {
            return url
        }
        return nil
    }

    /// Decoded byte count when known.
    public var byteCount: Int? {
        if case .cachedFile(_, let byteCount) = self {
            return byteCount
        }
        return nil
    }
}

File: android/ccda-engine/src/main/kotlin/com/ccdakit/engine/Models/Media/CCDAMediaPayload.kt
package com.ccdakit.engine

import java.io.File

/** Media payload storage. */
sealed class CCDAMediaPayload {
    /** Payload decoded and written to disk. */
    data class CachedFile(val file: File, val byteCount: Int) : CCDAMediaPayload()
    /** Payload kept as base64 text, usually when caching is disabled or decoding fails. */
    data class InlineBase64(val value: String) : CCDAMediaPayload()
    /** Payload was not available or could not be represented. */
    data object Unavailable : CCDAMediaPayload()
}
